### PR TITLE
Backtester rpc: start, end & interval overrides

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description

Just a small little update from another branch that isn't related to this feature. So ported it across. If you have a cool strategy, rather than editing the file, can now override the start, end and candle intervals with the request.

Also in my previous PR I somehow didn't commit the .proto changes, so now I've done it here.

Fixes # (issue)

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix 

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Coverage of new functionality in grpcserver.go

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backtester/engine/grpcserver.go [hunk 1]
```diff
@@ -210,6 +210,28 @@ func (s *GRPCServer) ExecuteStrategyFromFile(_ context.Context, request *btrpc.E
...
 
+	io64 := int64(request.IntervalOverride)
+	if io64 > 0 {
+		if io64 < gctkline.FifteenSecond.Duration().Nanoseconds() {
+			return nil, fmt.Errorf("%w %v", gctkline.ErrInvalidInterval, request.IntervalOverride)
+		}
+		cfg.DataSettings.Interval = gctkline.Interval(request.IntervalOverride)
+	}
+
+	if request.StartTimeOverride != nil && !request.StartTimeOverride.AsTime().IsZero() {
```
Comment (shazbert) [inline backtester/engine/grpcserver.go:221]: request.StartTimeOverride.AsTime() calls `time.Unix` with zero values so this is always false even if nil.

File: backtester/engine/grpcserver.go [hunk 2]
```diff
@@ -210,6 +210,28 @@ func (s *GRPCServer) ExecuteStrategyFromFile(_ context.Context, request *btrpc.E
...
+		if cfg.DataSettings.DatabaseData != nil {
+			cfg.DataSettings.DatabaseData.StartDate = request.StartTimeOverride.AsTime()
```
Comment (shazbert) [inline backtester/engine/grpcserver.go:223]: code coverage is missing here and below.

File: exchanges/binanceus/binanceus.go [hunk 3]
```diff
@@ -913,8 +913,7 @@ func (bi *Binanceus) GetSubaccountTransferHistory(ctx context.Context,
 	startTimeT := time.UnixMilli(int64(startTime))
 	endTimeT := time.UnixMilli(int64(endTime))
 
-	hundredDayBefore := time.Now()
-	hundredDayBefore.Sub(time.UnixMilli(int64((time.Hour * 24 * 10) / time.Millisecond)))
+	hundredDayBefore := time.Now().Add(-time.Hour * 24 * 100).Truncate(time.Hour)
```
Comment (shazbert) [inline exchanges/binanceus/binanceus.go:916]: 🔥

**Review by shazbert — CHANGES_REQUESTED** [review]
Thank you sir this looks better and cool that we can override times, just some minor nits.

File: backtester/engine/grpcserver.go (hunk 1 above)
Comment (gloriousCode) [reply backtester/engine/grpcserver.go:221]:
```go
func TestNil(t *testing.T) {
	t.Parallel()
	hello := timestamppb.New(time.Now())
	hello = nil
	if hello.AsTime().Unix() == 0 {
		t.Log("hi")
	}
}
```
Thanks! Didn't know!

**Review by shazbert — APPROVED** [review]
tACK 🔥

**Review by thrasher- — APPROVED** [review]
ACKnowledged

**Pull request merged**